web/src/user_group_edit.ts

```diff
@@ -1415,6 +1417,12 @@ export function change_state(
     if (/\d+/.test(section)) {
         const group_id = Number.parseInt(section, 10);
         const group = user_groups.get_user_group_from_id(group_id);
+        const group_visibility = group.deactivated
+            ? FILTERS.DEACTIVATED_GROUPS
+            : FILTERS.ACTIVE_GROUPS;
```

Comment:
One thing that we can do here is that whether we should show the filter in localstorage if that filter includes the group. Like if the last used filter was "Active and Deactivated", then we can just keep that filter. But am not sure, can be discussed on CZO first.